src/restruct_poc/solver/assemble_constraint_force.hpp

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <Kokkos_Core.hpp>
+
+#include "constraints.hpp"
+
+#include "src/restruct_poc/math/quaternion_operations.hpp"
+#include "src/restruct_poc/math/vector_operations.hpp"
+#include "src/restruct_poc/types.hpp"
+
+namespace openturbine {
+
+struct AssembleConstraintForce {
+    Kokkos::View<Constraints::DeviceData*>::const_type data;
+    View_N::const_type control;
+    View_Nx7::const_type node_u;
+    View_N R_system;
+
+    KOKKOS_FUNCTION
+    void operator()(const int i_constraint) const {
+        const auto& cd = data(i_constraint);
+
+        // Applies the torque from a revolute joint constraint to the system residual
+        if (cd.type == ConstraintType::kRevoluteJoint && cd.control) {
```

Comment:
There's no need to check cd.control here (plus, it should just be deleted from the struct, as I recommended in another comment).